Cooper/Controllers/RegistrationController.cs

```diff
@@ -14,32 +14,70 @@ namespace Cooper.Controllers
     [Route("api/registration")]
     public class RegistrationController : Controller
     {
-
         UserRepository userRepository;
-
-        public RegistrationController(IJwtHandlerService jwtHandler, IConfigProvider configProvider)
+        ISmtpClient smtpClient;
+        public RegistrationController(IJwtHandlerService jwtHandler, IConfigProvider configProvider, ISmtpClient smtpClient)
         {
             userRepository = new UserRepository(jwtHandler, configProvider);
+            this.smtpClient = smtpClient;
         }
 
         [HttpPost]
         [ProducesResponseType(200)]
         [ProducesResponseType(201)]
         public IActionResult Post([FromBody]UserRegistration user, string Password)
         {
-            bool nicknameExists = userRepository.IfNicknameExists(user.Nickname);
-            bool emailExists = userRepository.IfEmailExists(user.Email);
+            // TODO: send the proper explanation for bad-request.
+
+            if (!ModelState.IsValid) return BadRequest(ModelState);
```

Comment:
multiple returns

Reply:
Because, there will be different errors.

Reply:
IActionResult result = null;
If (...)
{
result = BadRequest(...)
}
else
{
.....
}
...
...
return result;

According to codestyle, all methods should have only one "return"